### PR TITLE
#2302 add serialization property IsPKSimModule

### LINKED ISSUE #2302: "Create proccess rate" property should not change a PK-Sim module to an extension module
Same as https://github.com/Open-Systems-Pharmacology/MoBi/issues/1443

### PR DESCRIPTION
**Pull request by benjaperez1983**
Fixes #2302

# Description

Add serialization property IsPKSimModule so that it can set on PkSim export and used in Mobi Import and change.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

**Comment by msevestre** [comment]
Can you add the number being fixed please ?
Otherwise the issue isn’t closed automatically

File: src/OSPSuite.Core/Domain/Module.cs [hunk 1]
```diff
@@ -36,15 +36,14 @@ public enum MergeBehavior
...
-
       private T buildingBlockByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().SingleOrDefault();
       private IReadOnlyList<T> buildingBlocksByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().ToList();
 
       /// <summary>
       ///    Module is a PKSim module if created in PKSim and the module version
       ///    matches the version when it was first imported
       /// </summary>
-      public bool IsPKSimModule => !string.IsNullOrEmpty(PKSimVersion) && Equals(Version, ModuleImportVersion);
+      public bool IsPKSimModule { get; set; }
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Module.cs:46]: Hum.  Why is this change required? We may end up with version not matching and the flag set to true

Comment (benjaperez1983) [reply src/OSPSuite.Core/Domain/Module.cs:46]: yeah well, from this story we are going to decouple the version from the IsPkSimModule, meaning that the version will be only for showing the simulation on Sync or not, but the PkSimModule will be just reflected by this property.
The reason for this, is that now, there are some propeties (the one mentioned on the original story) that should not change the Type of module (AKA IsPkSimModule) but should change the version.

Comment (msevestre) [reply src/OSPSuite.Core/Domain/Module.cs:46]: Well then update the property comment because it’s wrong

Comment (msevestre) [reply src/OSPSuite.Core/Domain/Module.cs:45]: Comment is misleading.  If the module was created in PKSim and something was changed then the flag is false but it was stjlll created in PK-Sim.   I don’t know what the first issue is about so I am not sure what constitute a PK-sim module 
@PavelBal can maybe explain

Comment (benjaperez1983) [reply src/OSPSuite.Core/Domain/Module.cs:45]: well would you say this comment is ok?
"Module is a PKSim module at the time of creation in PKSim, can be updated by changing some properties on Mobi" ?

**Comment by PavelBal** [comment]
>  I don’t know what the first issue is about so I am not sure what constitute a PK-sim module

The original issue is about changing this property:

image

Should not convert a PK-Sim module to an extension module.

File: src/OSPSuite.Core/Domain/Module.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Domain/Module.cs:45]: Yeah, the issue is that some commands in MoBi will change the building block but should not convert the module to extension.

We are going to need the commands themselves to know if they should unset this flag.

File: tests/OSPSuite.Core.Tests/Domain/ModuleSpecs.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
 using FakeItEasy;
+using NUnit.Framework;
```
Comment (rwmcintosh) [inline tests/OSPSuite.Core.Tests/Domain/ModuleSpecs.cs:4]: can you check if this is really needed? Seems it should not be.

**Review by rwmcintosh — APPROVED** [review]

**Comment by Yuri05** [comment]
I assume, with this change PK-Sim has to set `module.IsPKSimModule = true` before exporting a module?
Do we have an issue for this in the PK-sim repo?

**Comment by benjaperez1983** [comment]
> I assume, with this change PK-Sim has to set `module.IsPKSimModule = true` before exporting a module? Do we have an issue for this in the PK-sim repo?

We do this is the issue:
github.com/Open-Systems-Pharmacology/PK-Sim/issues/2953

**Review by Yuri05 — APPROVED** [review]
